core/src/main/java/com/linecorp/armeria/internal/common/util/TemporaryThreadLocals.java

```diff
@@ -148,28 +163,25 @@ public StringBuilder stringBuilder() {
         }
     }
 
-    /**
-     * Switches the internal representation of the specified {@link StringBuilder} from LATIN1 to UTF16,
-     * so that character operations do not have performance penalty.
-     */
-    @SuppressWarnings("checkstyle:AvoidEscapedUnicodeCharacters")
-    private static StringBuilder inflate(StringBuilder stringBuilder) {
-        stringBuilder.append('\u0100');
-        stringBuilder.setLength(0);
-        return stringBuilder;
+    private void lock() {
+        checkState(!lock, "Cannot be acquired before releasing the resource");
```

Comment:
Could use assertion in this case because the failure to lock would be a bug.

Reply:
SGTM! Updated.